MaxText/train.py

```diff
@@ -174,18 +174,20 @@ def train_step(model, config, state, data, dropout_rng):
     data[k] = v[:config.global_batch_size_to_train_on,:]
 
   def loss_fn(params):
+    padding_mask = data['targets_segmentation'] != 0
     logits, intermediate_outputs = model.apply({'params': params},
                          data['inputs'],
                          data['inputs_position'],
                          decoder_segment_ids=data['inputs_segmentation'],
+                         padding_mask=padding_mask,
                          enable_dropout=config.enable_dropout,
                          rngs={'dropout': rng1, 'aqt': aqt_rng}, mutable='intermediates')
     one_hot_targets = jax.nn.one_hot(data['targets'], config.vocab_size)
     xent, _ = max_utils.cross_entropy_with_logits(logits, one_hot_targets, 0.0)
     xent = nn.with_logical_constraint(xent, ('activation_batch', 'activation_length'))
     # Mask out paddings at the end of each example.
-    xent = xent * (data['targets_segmentation'] != 0)
-    return jnp.sum(xent)/jnp.sum((data['targets_segmentation'] != 0)), intermediate_outputs
+    xent = xent * padding_mask
+    return jnp.sum(xent)/jnp.size(padding_mask), intermediate_outputs
```

Comment:
Shouldn't we not normalize by jnp.size(padding_mask) but instead by the nonzeros? I think this is bringing the misnormalization back.

Reply:
`padding_mask` is equal to nozeros, calculation are equivalent here. Since `padding_mask` is not needed will roll back.